### ISSUE TITLE
postcode 22-120 (Poland) not found

### ISSUE THREAD

**Issue by zawadaa** [osm-search/Nominatim#1286]
Search: Sending request: 
`GET /search?postalcode=22-120&country=Polska&limit=5&format=json HTTP/1.1`

Response:
`"[{"place_id":"148079706","licence":"Data [0xc2][0xa9] OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright","osm_type":"way","osm_id":"311466871","boundingbox":["49.8597334","49.8598682","19.3417086","19.3421914"],"lat":"49.8597956","lon":"19.3419446103347","display_name":"200-lecia, Andrych[0xc3][0xb3]w, gmina Andrych[0xc3][0xb3]w, powiat wadowicki, ma[0xc5][0x82]opolskie, 34-120, RP","class":"place","type":"postcode","importance":0.225}]"`

Should be:
` "50.9131881" "23.5417781"`
more or less

**Comment by mtmail** [comment]
I see the issue. Just putting it in different words:

https://nominatim.openstreetmap.org/search.php?q=22-120%2C+poland should return a postcode. 

Other nearby postcodes work:
https://nominatim.openstreetmap.org/search.php?q=22-121%2C+poland
https://nominatim.openstreetmap.org/search.php?q=22-122%2C+poland

There is at least one place in Poland has `addr:postcode=22-120` set, e.g. https://www.openstreetmap.org/way/454709228 and there was no recent change. That place is in Nominatim database https://nominatim.openstreetmap.org/details.php?osmtype=W&osmid=454709228

The `location_postcode` table has an entry and the calculated centre point is inside Poland (so we can rule out miscalculation)

```
nominatim=# select country_code, postcode, st_x(geometry), st_y(geometry) from location_postcode where country_code='pl' and postcode like '22-12%' order by postcode;
 country_code | postcode |       st_x       |       st_y
--------------+----------+------------------+------------------
 pl           | 22-120   | 23.5537450641604 | 50.9230575744326
 pl           | 22-121   | 23.5217429118755 | 51.0327019031405
 pl           | 22-122   | 23.5009835212044 | 50.9795909394124
```

**Comment by lonvia** [comment]
The minus is removed during the normalisation but still there in the location_postcode table. Consequently there is no match. Kind of the same problem as #927.

**Comment by lonvia** [comment]
Turns out this was the same issue as #1953. There is a postcode '22-120' and '22 120' and one did overshadow the other in the word table. As a consequence the postcode could be found in only one form. This is now fixed with #1960.
